site/lib/src/components/client/archive_table.dart

```diff
@@ -0,0 +1,229 @@
+        table(
+          id: 'downloads-$os-$channel',
+          classes: 'table table-striped',
+          [
+            thead([
+              tr([
+                th([text('Flutter version')]),
+                th([text('Architecture')]),
+                th([text('Ref')]),
+                th(classes: 'date', [text('Release Date')]),
+                th([text('Dart version')]),
+                th([text('Provenance')]),
+              ]),
+            ]),
+            tbody([
+              if (isLoading)
+                tr(classes: 'loading', [
+                  td(attributes: {'colspan': '6'}, [text('Loading...')]),
+                ])
+              else if (error != null)
+                tr(classes: 'error', [
+                  td(
+                    attributes: {'colspan': '6'},
+                    [
+                      text(
+                        'Failed to load releases. Refresh page to try again.',
+                      ),
+                    ],
+                  ),
+                ])
+              else
+                for (final release in releases)
+                  tr([
+                    td([
+                      a(href: release.url, [
+                        text(release.version),
+                      ]),
+                    ]),
+                    td([
+                      span([text(release.architecture)]),
+                    ]),
+                    td([
+                      span(classes: 'git-hash', [text(release.hash)]),
+                    ]),
+                    td(classes: 'date', [
+                      text(
+                        release.releaseDate.toLocaleDateString(),
+                      ),
+                    ]),
+                    td([
+                      span([text(release.dartSdkVersion)]),
+                    ]),
+                    td([
+                      buildProvenanceLink(release),
+                    ]),
+                  ]),
+            ]),
+          ],
+        ),
+      ]),
+    ]);
+  }
+
+  static const baseUrl =
+      'https://storage.googleapis.com/flutter_infra_release/releases/';
+
+  static final windowsCutoff = Date.parse('4/3/2023');
+  static final otherOsCutoff = Date.parse('12/15/2022');
+
+  Component buildProvenanceLink(Release release) {
+    final dateValue = release.releaseDate.valueOf();
+
+    if (os == 'windows' && dateValue < windowsCutoff) {
+      // Provenance not available before 4/3/2023 for Windows
+      return span([text('-')]);
+    } else if (dateValue < otherOsCutoff) {
+      // Provenance not available before 12/15/2022 for macOS and Linux
+      return span([text('-')]);
+    }
+
+    final extension = os == 'linux' ? 'tar.xz' : 'zip';
+    return a(
+      href:
+          '$baseUrl$channel/$os/flutter_${os}_${release.version}-$channel.$extension.intoto.jsonl',
+      target: Target.blank,
+      [text('Attestation bundle')],
+    );
+  }
+}
+
+/// The JavaScript Date object.
```

Comment:
Good idea to use this rather than `package:intl` for this focused use case. Would it make sense to do the same for the dart.dev archive as well?

Reply:
I will take a look